target/linux/lantiq/dts/EASY220.dtsi

```diff
@@ -0,0 +1,295 @@
+			lantiq,function = "pci";
+		};*/
+		/*led {
+			lantiq,groups = "led";
+			lantiq,function = "led";
+		};*/
+		conf_in {
+			lantiq,pins = "io39"; /* exin3 */
+			lantiq,pull = <2>;
+		};
+		conf_out {
+			lantiq,pins =  // "io21", "io38", /* pcie */
+//					"io12", "io15", /* annexa, annexb */
+					"io0", "io1", "io8", "io11", "io33", "io36"; /* led io34, io35 */
+			lantiq,open-drain = <0>;
+			lantiq,pull = <0>;
+			lantiq,output = <1>;
+		};
+	};
+	pins_spi_default: pins_spi_default {
+		spi_in {
+			lantiq,groups = "spi_di";
+			lantiq,function = "spi";
+		};
+		spi_out {
+			lantiq,groups = "spi_do", "spi_clk", "spi_cs4";
+			lantiq,function = "spi";
+			lantiq,output = <1>;
+		};
+	};
+};
+
+&spi {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pins_spi_default>;
+
+	status = "ok";
+
+	m25p80@4 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "winbond,w25q64", "jedec,spi-nor";
+		reg = <4 0>;
+		linux,modalias = "m25p80", "w25q64";
+		linux,mtd-name = "ltq_sflash";
+		spi-max-frequency = <10000000>;
+		m25p,fast-read;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				reg = <0x0 0x20000>;
+				label = "uboot";
+				read-only;
+			};
+
+			partition@20000 {
+				reg = <0x20000 0x10000>;
+				label = "gphyfirmware";
+				read-only;
+			};
+
+			partition@30000 {
+				reg = <0x30000 0x190000>;
+				label = "kernel";
+				read-only;
+			};
+
+			partition@1c0000 {
+				reg = <0x1c0000 0x80000>;
+				label = "firmware";
+				read-only;
+			};
+
+			partition@240000 {
+				reg = <0x240000 0x500000>;
+				label = "rootfs";
+				read-only;
+			};
+
+			partition@imgalias {
+				reg = <0x30000 0x710000>;
+				label = "image";
+				read-only;
+			};
+
+			partition@0x740000 {
+				reg = <0x740000 0x10000>;
+				label = "dniconfig";
+				read-only;
+			};
+
+			partition@0x750000 {
+				reg = <0x750000 0x10000>;
+				label = "log";
+				read-only;
+			};
+
+			partition@0x760000 {
+				reg = <0x760000 0x80000>;
+				label = "language";
+				read-only;
+			};
+
+			partition@0x7e0000 {
+				reg = <0x7e0000 0x10000>;
+				label = "sysconfig";
+				read-only;
+			};
+
+			partition@0x7f0000 {
+				reg = <0x7f0000 0x2000>;
+				label = "ubootconfig";
+				read-only;
+			};
+
+			partition@0x7f2000 {
+				reg = <0x7f2000 0x1000>;
+				label = "ART";
+				read-only;
+			};
+
+			partition@0x7f3000 {
+				reg = <0x7f3000 0x1000>;
+				label = "pot";
+				read-only;
+			};
+
+			partition@0x7f4000 {
+				reg = <0x7f4000 0xc000>;
+				label = "ret";
+				read-only;
+			};
+		};
+	};
+};
+
+&pcie0 {
+	status = "disabled";
+};
+
+&eth0 {
+//	interrupts = <73 72 95 96>;
+
+	lan: interface@0 {
+		compatible = "lantiq,xrx200-pdi";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		reg = <0>;
+//		reg = <2>;
+		mtd-mac-address = [ 00 11 22 33 44 55 ];
+//		mtd-mac-address = [ 00 11 22 33 44 57 ];
+//		mtd-mac-address = <&romfile 0xf100>;
```

Comment:
is it possible to get the mac address from the art partition? The art partition might contain a mac address that only needs to incremented/decremented. Please have a look at the other dts files for examples.